### PR TITLE
Bugfix/getuploadcount

### LINKED ISSUE #3048: IllegalStateException while fetching uploadCount in OkHttpJsonApiClient
**Summary:** 
Fetching upload count fails throwing exception mentioned in the below logs

**System logs:**

```
Fetching upload count failed
    java.lang.IllegalStateException: closed
        at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:407)
        at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:401)
        at okhttp3.internal.Util.bomAwareCharset(Util.java:471)
        at okhttp3.ResponseBody.string(ResponseBody.java:175)
        at fr.free.nrw.commons.mwapi.OkHttpJsonApiClient.lambda$getUploadCount$0$OkHttpJsonApiClient(OkHttpJsonApiClient.java:102)
        at fr.free.nrw.commons.mwapi.-$$Lambda$OkHttpJsonApiClient$9ZpGX0U5Zngts_jcRCFls7FtEIM.call(lambda)
        at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
        at io.reactivex.Single.subscribe(Single.java:3438)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:762)
```

 
**Commons app version:** 

Current master

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by ashishkumar468**
**Description (required)**

Fixes #3048 

What changes did you make and why?
Extracted out response.body().string() to a variable and used that for further references.
**Tests performed (required)**

Tested betaDebug on Pixel Api 27

### CODE REVIEW

**Comment by neslihanturan** [comment]
Hi @ashishkumar468 I get this error:

```
    java.lang.NumberFormatException: For input string: "okhttp3.internal.http.RealResponseBody@86d3e48"
        at java.lang.Integer.parseInt(Integer.java:608)
        at java.lang.Integer.parseInt(Integer.java:643)
        at fr.free.nrw.commons.mwapi.OkHttpJsonApiClient.lambda$getUploadCount$0$OkHttpJsonApiClient(OkHttpJsonApiClient.java:106)
        at fr.free.nrw.commons.mwapi.-$$Lambda$OkHttpJsonApiClient$9ZpGX0U5Zngts_jcRCFls7FtEIM.call(Unknown Source:4)
        at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
        at io.reactivex.Single.subscribe(Single.java:3438)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
```

Is this expected?

**Comment by ashishkumar468** [comment]
It is not, I will look into it, thanks for testing this out @neslihanturan 
 :-)

File: app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java [hunk 1]
```diff
@@ -98,8 +99,12 @@ public OkHttpJsonApiClient(OkHttpClient okHttpClient,
         return Single.fromCallable(() -> {
             Response response = okHttpClient.newCall(request).execute();
             if (response != null && response.isSuccessful()) {
-                if(!TextUtils.isEmpty(response.body().string().trim())){
-                    return Integer.parseInt(response.body().string().trim());
+                ResponseBody responseBody = response.body();
+                if(null!=responseBody) {
+                    String responseBodyString = responseBody.toString();
```
[the remark this answers is not in the document]
Comment (ashishkumar468) [reply app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java]: Done

File: app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java [hunk 2]
```diff
@@ -98,8 +99,12 @@ public OkHttpJsonApiClient(OkHttpClient okHttpClient,
...
+                    if (!TextUtils.isEmpty(responseBodyString.trim())) {
+                        return Integer.parseInt(responseBodyString.trim());
```
[the remark this answers is not in the document]
Comment (ashishkumar468) [reply app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java]: Added a catch for NumberFormatException

**Comment by ashishkumar468** [comment]
@neslihanturan I have handled the exception as well. Please look into it now :-)

**Pull request merged**